launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -107,6 +119,30 @@ public static void setConfig(String name, String value) {
     launcherConfig.put(name, value);
   }
 
+  /**
+   * Specifies that Spark Application be stopped if current process goes away.
+   * It tries stop/kill Spark Application if {@link LauncherServer} goes away.
+   *
+   * @since 2.2.0
+   * @return This launcher.
+   */
+  public SparkLauncher autoShutdown() {
```

Comment:
This is inconsistent with `launchAsThread` which takes a boolean. I think adding the parameter is better.